code/TaxonomyTerm.php

```diff
@@ -125,6 +126,29 @@ public function onBeforeDelete()
             $term->delete();
         }
     }
+    /**
+     *  Set the URL segment allowing filtering by url slug
+     *
+     *  {@inheritDoc}
+     */
+    public function onBeforeWrite() {
+
+        parent::onBeforeWrite();
+
+        // Write the URLSegment to allow for Taxonomy navigation only if one is not supplied
+        if ($this->Name && $this->URLSegment == null)
+        {
+            $this->URLSegment = singleton('Sitetree')->generateURLSegment($this->Name);
+
+            // Ensure that this object has a non-conflicting URLSegment value.
+            $count = 2;
+            while ( TaxonomyTerm::get()->filter(['URLSegment' => $this->URLSegment])->exclude(array('ID' => $this->ID))->exists() )
```

Comment:
looping through a while to assign a suffix number if duplicate (using same code from SiteTree).